tests/characters/control/test_web_control.py

```diff
@@ -99,12 +99,15 @@ def test_alice_character_control_revoke(alice_web_controller_test_client, federa
     response = alice_web_controller_test_client.put('/grant', data=json.dumps(grant_request_data))
     assert response.status_code == 200
 
+    policy_credential = PolicyCredential.from_json(
+                                response.json['result']['policy_credential'])
```

Comment:
This style of formatting (odd newline placement, IMO) is throughout your PR - are running black?

Reply:
That's the way to please PEP8 gods with the 80-character rule!

Reply:
in this case (and in many) you can get a shorter, more readable block by adding a variable:

```
    credential_json = response.json['result']['policy_credential']
    policy_credential = PolicyCredential.from_json()
```